src/tiledb/cloud/file/indexing.py

```diff
@@ -0,0 +1,405 @@
+        to specify the worker Docker image.
+    :param extra_worker_modules: If `embeddings_generation_mode=BATCH` this can be
+        used to install extra pip package to the image.
+    :param driver_resources: If `embeddings_generation_driver_mode=BATCH` this can
+        be used to specify the driver resources.
+    :param driver_image: If `embeddings_generation_driver_mode=BATCH` this can be
+        used to specify the driver Docker image.
+    :param extra_driver_modules: If `embeddings_generation_driver_mode=BATCH` this
+        can be used to install extra pip package to the image.
+    :param max_tasks_per_stage: Number of maximum udf tasks per computation stage.
+    :param embeddings_generation_mode: TaskGraph execution mode for embeddings
+        generation.
+    :param embeddings_generation_driver_mode: TaskGraph execution mode for the
+        ingestion driver.
+    :param vector_indexing_mode: TaskGraph execution mode for the vector indexing.
+    :param index_update_kwargs: Extra arguments to pass to the index update job.
+        These can be any of the documented tiledb.vector_search.ingest method with the
+        exception of BATCH Embedding Resources (see next params):
+        https://tiledb-inc.github.io/TileDB-Vector-Search/documentation/reference/ingestion.html#tiledb.vector_search.ingestion.ingest
+        Also `files_per_partition: int` can be included (defaults to -1)
+    ## Vector Search BATCH Embedding Resources
+    ## These are only applicable if indexing update is executed in BATCH mode.
+    :param threads: Threads to be used in the Nodes, defaults to 16.
+    :param ingest_resources: Resources to request when performing vector ingestion.
+    :param consolidate_partition_resources: Resources to request when performing
+        consolidation of a partition.
+    :param copy_centroids_resources: Resources to request when performing copy
+        of centroids from input array to output array.
+    :param random_sample_resources: Resources to request when performing random
+        sample selection.
+    :param kmeans_resources: Resources to request when performing kmeans task.
+    :param compute_new_centroids_resources: Resources to request when performing
+        centroid computation.
+    :param assign_points_and_partial_new_centroids_resources: Resources to request
+        when performing the computation of partial centroids.
+    :param write_centroids_resources: Resources to request when performing the
+        write of centroids.
+    :param partial_index_resources: Resources to request when performing the
+        computation of partial indexing.
+    """
+
+    from tiledb.cloud.utilities import run_dag
+
+    # Argument preparation
+    DEFAULT_IMG_NAME = "vectorsearch"
+    worker_resources = worker_resources or {"cpu": "2", "memory": "8Gi"}
+    driver_resources = driver_resources or {"cpu": "1", "memory": "4Gi"}
+    ## Vector Search BATCH Embedding Resources:
+    index_update_kwargs.update(
```

Comment:
the named args form of this might make it easier to read:

```python
index_update_kwargs.update(
    ingest_resources=ingest_resources or ...,
    consolidate_partition_resources=...,
    ...,
)
```